test/flycheck-test.el

```diff
@@ -3563,11 +3563,17 @@ Why not:
    '(4 2 error "Syntax error, maybe a missing semicolon?"
        :checker processing)))
 
-(flycheck-ert-def-checker-test puppet-parser puppet parser-error
+; the puppet 4 and 3 tests are mutually exclusive due to one having column and the other not
+(flycheck-ert-def-checker-test puppet-parser puppet parser-error-puppet-4
   (flycheck-ert-should-syntax-check
    "language/puppet/parser-error.pp" 'puppet-mode
    '(3 9 error "Syntax error at '>'" :checker puppet-parser)))
 
+(flycheck-ert-def-checker-test puppet-parser puppet parser-error-puppet-3
+  (flycheck-ert-should-syntax-check
+   "language/puppet/parser-error2.pp" 'puppet-mode
+   '(4 nil error "Syntax error at 'helloagain'; expected '}'" :checker puppet-parser)))
```

Comment:
Please skip either test when the corresponding puppet version is not available.  In other words the test for Puppet 3 errors shouldn't run with Puppet 4 and vice versa.